json-schema-converter/src/test/java/io/confluent/connect/json/JsonSchemaDataTest.java

```diff
@@ -803,6 +806,88 @@ public void testToConnectRecordWithOptionalAndRequired() {
     checkNonObjectConversion(jsonSchemaData, expectedSchema, struct, schema, obj);
   }
 
+  @Test
+  public void testToConnectRecordWithOptionalReference() throws Exception {
+    String schemaString = "{\n"
+        + "  \"$schema\": \"http://json-schema.org/draft-04/schema#\",\n"
+        + "  \"title\": \"Event\",\n"
+        + "  \"type\": \"object\",\n"
+        + "  \"properties\": {\n"
+        + "    \"vehicle\": {\n"
```

Comment:
I wonder if we should rename these variable names (vehicle,vehicleDto etc). Its a customer schema and the code could be in opensource.

Reply:
This is not the customer schema, it's a stripped down version with very little info